opensrp-app/src/main/java/org/smartregister/CoreLibrary.java

```diff
@@ -43,6 +51,18 @@ public static CoreLibrary getInstance() {
     private CoreLibrary(Context contextArg, SyncConfiguration syncConfiguration) {
         context = contextArg;
         this.syncConfiguration = syncConfiguration;
+
+        initP2pLibrary();
+    }
+
+    private void initP2pLibrary() {
+        AllSharedPreferences allSharedPreferences = new AllSharedPreferences(getDefaultSharedPreferences(context.applicationContext()));
+        String username = allSharedPreferences.fetchRegisteredANM();
+
+        if (!TextUtils.isEmpty(username)) {
+            P2PLibrary.init(new P2PLibrary.Options(username
+                    , new P2PSyncAuthorizationService(allSharedPreferences.fetchDefaultTeamId(username))));
```

Comment:
allow OpenSRP application to pass an instance or child class of P2PSyncAuthorizationService